plasmapy/diagnostics/thomson.py

```diff
@@ -0,0 +1,275 @@
+    ne: u.m ** -3,
+    Te: u.K,
+    Ti: u.K,
+    fract: np.ndarray = np.ones(1),
+    ion_species: Union[str, List[str], Particle, List[Particle]] = "H+",
+    fluid_vel: u.m / u.s = np.zeros(3) * u.cm / u.s,
+    ion_vel: u.m / u.s = None,
+    probe_vec=np.array([1, 0, 0]),
+    scatter_vec=np.array([0, 1, 0]),
+) -> Tuple[Union[np.floating, np.ndarray], np.ndarray]:
+    r"""
+    Calculate the spectral density function for Thomson scattering of a
+    probe laser beam by a multi-species Maxwellian plasma.
+
+    This function calculates the spectral density function for Thomson
+    scattering of a probe laser beam by a plasma consisting of one or more ion
+    species and a neutralizing electron fluid:
+
+    .. math::
+        S(k,\omega) = 1 - \frac{2\pi}{k}
+        \bigg |1 - \frac{\chi_e}{\epsilon} \bigg |^2
+        f_{e0} \bigg (\frac{\omega}{k} \bigg ) +
+        \sum_i \frac{2\pi Z_i}{k}
+        \bigg |\frac{\chi_e}{\epsilon} \bigg |^2 f_{io}
+        \bigg ( \frac{\omega}{k} \bigg )
+
+    where:
+
+    .. math::
+        \epsilon = 1 + \chi_e + \sum_i \chi_i
+
+    In this function the electron and ion distribution functions
+    :math:`f_{e0}` and :math:`f_{i0}` are assumed to be Maxwellian, making this
+    function equivalent to Eq. 3.4.6 in Sheffield.
+
+    Parameters
+    ---------
+
+    wavelengths : astropy.units.Quantity ndarray (required)
+        Array of wavelengths over which the spectral density function
+        will be calculated, convertable to nm.
+
+    probe_wavelength : astropy.units.Quantity (required)
+        Wavelength of the probe laser convertable to nm
+
+    ne : astropy.units.Quantity, ndarray (required)
+        mean (0th order) electron density of all plasma components combined
+        convertable to cm^-3.
+
+    Te : astropy.units.Quantity (required)
+        Temperature of the electron component (convertable to either eV or K)
+
+    Ti : astropy.units.Quantity ndarray, shape [N] (required)
+        Temperature of each ion component (convertable to either eV or K)
+
+    fract : float ndarray, shape [N]
+        Fraction (by number) of the total number of ions made up by each ion
+        species. Must sum to 1.0. Default is a single ion species.
+
+    ion_species : str or Particle class list, shape [N]
+        A list of either instances of the Particle class or strings
+        interpretable by that class representing each ion species.
+        Default is ['H+'] corresponding to a single species of hydrogen ions.
+
+    fluid_vel : astropy.units.Quantity ndarray shape [3]
+        Velocity of the fluid or electron component in the rest frame in
+        units convertable to cm/s. Defaults to [0, 0, 0], representing
+        a stationary plasma.
+
+    ion_vel : astropy.units.Quantity ndarray, shape [N,3]
+        Velocity vectors for each ion population relative to the
+        fluid velocit in units convertable to cm/s. Defaults zero drift
+        for all specified ion species.
```

Comment:
```suggestion
    ion_vel : `~astropy.units.Quantity`, shape (N, 3), optional
        Velocity vectors for each ion population relative to the
        fluid velocity. (convertible to cm/s) Defaults zero drift
        for all specified ion species.
```

Reply:
@pheuer Do you have a reason for not adding this suggestion, or did it just get lost in the noise?

Reply:
@rocco8773  Oops no, just got lost amongst all the other commits